### PR TITLE
Feature/111 local storage

### PR DESCRIPTION
**Pull request by Mortadha-abderrhim**
Stores User's favorite experiences in a Room (Android local DB).
A favorite is defined as UserID, ExperienceID, and Experience in JSON format. The pair (UserID, ExperienceID) is the primary key.

### CODE REVIEW

**Review by aouame — APPROVED** [review]
It looks good, nj!
Reformatting all the code kind of made the pr heavier than what it is.

**Review by aouame — APPROVED** [review]

File: app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt [hunk 1]
```diff
@@ -1,118 +1,133 @@
...
+            user: PindexUser?,
+            roomFavoriteRepository: FavoriteRepository
+        ): Boolean {
+            if (exp == null || user == null)
+                return false
+            var exists = false
+            runBlocking {
+                exists = roomFavoriteRepository.checkExistence(user.uid, exp)
+            }
+            return exists
```
Comment (ToufiPF) [inline app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt:46]: I think it'd be clearer to just `return runBlocking { roomFavoritesRepository.checkExistence(user.uid, exp) }` here

File: app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt [hunk 2]
```diff
@@ -1,118 +1,133 @@
...
+                shouldCheckAddToFavorites(exp, user, favoriteRepository)
+            )
+        }
+    }
+
+    fun toggleAddToFavorite() {
+        experience.value?.let { exp ->
+            userRepo.user.value?.let { user ->
+                viewModelScope.launch {
+                    if (mShouldAddToFavoritesBeChecked.value == true)
```
Comment (ToufiPF) [inline app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt:108]: Same here, it'd be clearer to test `shouldAddToFavoritesBeChecked.value?`

**Review by ToufiPF — COMMENTED** [review]
I just have a few comments regarding the code style, but otherwise GJ !

File: app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt (hunk 1 above)
Comment (Mortadha-abderrhim) [reply app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt]: done

File: app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt (hunk 2 above)
Comment (Mortadha-abderrhim) [reply app/src/main/java/com/pindex/main/viewmodels/ExperienceViewModel.kt]: done

**Review by aouame — APPROVED** [review]

**Review by hadiyahya11 — APPROVED** [review]

**Pull request merged**